packages/plugin-team/src/encrypter.js

```diff
@@ -0,0 +1,57 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ * @private
+ */
+
+import {Encrypter} from '@ciscospark/plugin-conversation';
+
+/**
+ * Encrypts a team
+ * @param {Object} team
+ * @param {Encryption~Key|string} usableKey
+ * @returns {Promise} Resolves with the encrypted team
+ * @private
+ */
+Encrypter.prototype._encryptTeam = function _encryptTeam(team, usableKey) {
+  return Promise.resolve(usableKey || this.spark.encryption.getUnusedKey())
+    .then((key) => {
+      const promises = [];
+
+      if (team.displayName) {
+        promises.push(this.encryptProperty(team, `displayName`, key)
+          // TODO can assignEncryptionKeyUrl be done inside encryptProperty? (
+          .then(() => {
+            team.encryptionKeyUrl = key.keyUrl;
+          }));
+      }
+
+      if (team.summary) {
+        promises.push(this.encryptProperty(team, `summary`, key)
+          // TODO can assignEncryptionKeyUrl be done inside encryptProperty? (
+          .then(() => {
+            team.encryptionKeyUrl = key.keyUrl;
+          }));
+      }
+
+      if (team.kmsMessage) {
+        promises.push(this.encryptProperty(team, `kmsMessage`, key));
+      }
+
+      return Promise.all(promises)
+        .then(() => {
+          team.defaultActivityEncryptionKeyUrl = key.keyUrl;
+        });
+    });
+};
+
+/**
+ * Encrypts a summary
+ * @param {string} summary
+ * @param {Encryption~Key|string} key
+ * @private
+ * @returns {Promise} Resolves with the encrypted string
+ */
+Encrypter.prototype._encryptPropSummary = function _encryptPropSummary(summary, key) {
```

Comment:
Consider setting all these new functions using Object.assign, that way we'll get the collision handling that we don't have in e.g. SquaredAdapter.js